avm/res/storage/storage-account/main.bicep

```
@@ -408,12 +413,16 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2024-01-01' = {
         keyvaultproperties: !empty(customerManagedKey)
           ? {
               keyname: customerManagedKey!.keyName
-              keyvaulturi: cMKKeyVault!.properties.vaultUri
+              keyvaulturi: !isHSMKeyVault
+                ? 'https://${last(split((customerManagedKey.?keyVaultResourceId!), '/'))}${environment().suffixes.keyvaultDns}/'
```

```suggestion
              keyvaulturi: cMKKeyVault!.properties.vaultUri
```
just so that we don't forget